apps/sv/src/main/scala/org/lfdecentralizedtrust/splice/sv/store/db/DbSvDsoStore.scala

```diff
@@ -529,7 +529,10 @@ class DbSvDsoStore(
       } yield applyLimit(opName, totalCouponsLimit, result).map { case (round, batch) =>
         RoundBatch(
           round,
-          batch.map(cid => companionClass.toContractId(companion, cid.contractId)).toSeq,
+          batch
+            .take(totalCouponsLimit.limit)
```

Comment:
Why not filter this out directly in SQL?

Reply:
agree, all good if I just add that to https://github.com/hyperledger-labs/splice/issues/2106?